addOns/openapi/src/main/java/org/zaproxy/zap/extension/openapi/generators/BodyGenerator.java

```diff
@@ -67,86 +63,135 @@ public BodyGenerator(Generators generators) {
                 + jsonStr
                 + SYNTAX.get(Element.ARRAY_END);
     }
 
-    private String generateJsonPrimitiveValue(Property property) {
-        return dataGenerator.generateBodyValue("", property, new ArrayList<>());
+    private String generateJsonPrimitiveValue(Schema<?> schema) {
+        return dataGenerator.generateBodyValue("", schema);
     }
 
-    private String generateJsonObjectString(String name, List<String> refs) {
-        StringBuilder json = new StringBuilder();
-        json.append(SYNTAX.get(Element.OBJECT_BEGIN));
-        boolean isFirst = true;
-        Map<String, Property> map = modelGenerator.getProperty(name);
-        if (map != null) {
-            for (Map.Entry<String, Property> property : map.entrySet()) {
-                if (isFirst) {
-                    isFirst = false;
-                } else {
-                    json.append(SYNTAX.get(Element.OUTER_SEPARATOR));
-                }
-                json.append(SYNTAX.get(Element.PROPERTY_CONTAINER));
-                json.append(property.getKey());
-                json.append(SYNTAX.get(Element.PROPERTY_CONTAINER));
-                json.append(SYNTAX.get(Element.INNER_SEPARATOR));
-                String value;
-                if (dataGenerator.isSupported(property.getValue().getType())) {
-                    value =
-                            dataGenerator.generateBodyValue(
-                                    property.getKey(), property.getValue(), refs);
-                } else {
-                    if (property.getValue() instanceof RefProperty) {
-                        value =
-                                generate(
-                                        ((RefProperty) property.getValue()).getSimpleRef(),
-                                        false,
-                                        refs);
-                    } else {
-                        value =
-                                generators
-                                        .getValueGenerator()
-                                        .getValue(
-                                                property.getKey(),
-                                                property.getValue().getType(),
-                                                generate(
-                                                        property.getValue().getName(),
-                                                        false,
-                                                        refs));
-                    }
-                }
-
-                json.append(value);
+    private Schema<?> resolveComposedSchema(ComposedSchema schema) {
+
+        if (schema.getOneOf() != null) {
+            return schema.getOneOf().get(0);
+        } else if (schema.getAnyOf() != null) {
+            return schema.getAnyOf().get(0);
+        }
+        // should not be reached
+        // allOf schema is resolved by the parser
+        LOG.error("Unknown composed schema type");
```

Comment:
Worth including the `schema` in the message (just in case it provides mode info).